kernel/fdtable.c

```diff
@@ -570,3 +570,30 @@ int myst_fdtable_list(const myst_fdtable_t* fdtable)
 
     return ret;
 }
+
+long myst_fdtable_sync(myst_fdtable_t* fdtable)
+{
+    long ret = 0;
+
+    if (!fdtable)
+        ERAISE(-EINVAL);
+
+    myst_spin_lock(&fdtable->lock);
+    {
+        for (int i = 0; i < MYST_FDTABLE_SIZE; i++)
+        {
+            const myst_fdtable_entry_t* entry = &fdtable->entries[i];
+
+            if (entry->type == MYST_FDTABLE_TYPE_FILE)
+            {
+                myst_fs_t* fs = entry->device;
+                fs->fs_sync(fs);
```

Comment:
For hostfs, the subsequent calls after the first one will be redundant.

Reply:
An alternate approach could be to call sync once on the host, if there is atleast one hostfs file open.

Reply:
I was thinking there could be multiple instances of hostfs. Like one hostfs rootfs plus several mounted hostfs. Let me think about if there is a way to distinguish those hostfs.